### PR TITLE
Returning a 404 error if the result Optional is empty

### PR DESCRIPTION
**Pull request by mmchougule**
Issue #17 
If the user doesn't exist, the performed value Optional is empty then return a 404 error instead of a 500.

### CODE REVIEW

File: src/main/g8/build.sbt [hunk 1]
```diff
@@ -1,5 +1,5 @@
-lazy val akkaHttpVersion = "$akka_http_version$"
-lazy val akkaVersion    = "$akka_version$"
+lazy val akkaHttpVersion = "10.0.11"
+lazy val akkaVersion    = "2.5.11"
```
Comment (ktoso) [inline src/main/g8/build.sbt]: Pretty sure this change should not be done; it's a template after all -- those numbers are getting replaced during generation of project from template

Comment (mmchougule) [reply src/main/g8/build.sbt]: Sorry, it looked like the Travis CI build was failing on my pull request to fix a 404 error. Reverting.

**Comment by raboof** [comment]
I think the travis errors are due to a major outage at Bintray, see http://status.bintray.com/

File: src/main/g8/src/main/java/com/lightbend/akka/http/sample/UserRoutes.java [hunk 2]
```diff
@@ -70,8 +70,13 @@ private Route getUser(String name) {
           return rejectEmptyResponse(() ->
             onSuccess(
               () -> maybeUser,
-              performed ->
-                complete(StatusCodes.OK, (User) performed.get(), Jackson.<User>marshaller())
+              performed -> {
+                  if (performed.isPresent())
+                    return complete(StatusCodes.OK, performed.get(), Jackson.marshaller());
```
Comment (raboof) [inline src/main/g8/src/main/java/com/lightbend/akka/http/sample/UserRoutes.java]: Nice that this apparently doesn't need those explicit types after all :+1:

**Review by raboof — APPROVED** [review]
Nice, thanks - let's wait until the problems at bintray are resolved so we can verify the build

File: src/main/g8/src/main/java/com/lightbend/akka/http/sample/UserRoutes.java (hunk 2 above)
Comment (mmchougule) [reply src/main/g8/src/main/java/com/lightbend/akka/http/sample/UserRoutes.java]: Yes! :)

**Review by raboof — COMMENTED** [review]
I guess we can remove the `rejectEmptyResponse` now as well, then?

**Comment by mmchougule** [comment]
That's true. Return a user if exists or not found. Removed ```rejectEmptyResponse``` and tests pass.

**Review by raboof — APPROVED** [review]

**Pull request merged**